tests/test_git.py

```diff
@@ -100,6 +102,51 @@ def test_apply_with_base_and_brackets(self):
         self.assertEqual(s.get_property("git.url"),
                          self.repo + " patchew/20160628014747.20971-2-famz@redhat.com")
 
+    def git_log(self, template):
+        return re.sub(r'XXXXXX/repo', self.repo, template).encode()
+
+    def test_rest_need_apply(self):
+        resp = self.apply_and_retrieve("0013-foo-patch.mbox.gz", self.p.id,
+                                       "20160628014747.20971-1-famz@redhat.com")
+        self.assertEqual(resp.data['results']['git']['status'], 'pending')
+
+    def test_rest_apply_failure(self):
+        self.cli_import("0014-bar-patch.mbox.gz")
+        self.do_apply()
+        resp = self.api_client.get(self.REST_BASE + 'series/?q=project:QEMU')
+        self.assertEqual(resp.data['results'][0]['is_complete'], True)
+        self.assertEqual(resp.data['results'][0]['results']['git']['status'], 'failure')
+        self.assertEqual('repo' in resp.data['results'][0]['results']['git'], False)
+        self.assertEqual('tag' in resp.data['results'][0]['results']['git'], False)
+        log = self.client.get(resp.data['results'][0]['results']['git']['log_url'])
+        expected_log = self.git_log("""Switched to a new branch '20160628014747.20971-2-famz@redhat.com'
+Applying: bar patch
+error: sha1 information is lacking or useless (foo).
+error: could not build fake ancestor
+Patch failed at 0001 bar patch
+The copy of the patch that failed is found in: .git/rebase-apply/patch
+When you have resolved this problem, run "git am --continue".
+If you prefer to skip this patch, run "git am --skip" instead.
+To restore the original branch and stop patching, run "git am --abort".
+Failed to apply patch:
+[Qemu-devel] [PATCH] bar patch""")
```

Comment:
I think the actual error message depends on the git version. I'm getting this on Fedora (git version 2.14.3):

```
Switched to a new branch '20160628014747.20971-2-famz@redhat.com'
Branch 20160628014747.20971-2-famz@redhat.com set up to track local branch test by rebasing.
Applying: bar patch
error: sha1 information is lacking or useless (foo).
error: could not build fake ancestor
Patch failed at 0001 bar patch
The copy of the patch that failed is found in: .git/rebase-apply/patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
Failed to apply patch:
[Qemu-devel] [PATCH] bar patch
```

I.e. there is one extra line in the middle:

```
Branch 20160628014747.20971-2-famz@redhat.com set up to track local branch test by rebasing.
```
Maybe simply remove the `self.assertEquals(log.content, expected_log)` check?

Reply:
I'll change it to a status code test (must be 200).